### ISSUE TITLE
[BMG][PyTorch] Training performance regressions with Triton 3.8.0 vs 3.7.2 on B60 Linux

### ISSUE THREAD

**Issue by mengfei25** [intel/intel-xpu-backend-for-triton#7511]
## Description

After upgrading Triton from **3.7.2** to **3.8.0** (`3.8.0+git9810c442`), a large number of models show **training performance regressions** on B60 Linux. Testing against the latest release **3.8.0 (`6c7855c1`)** shows most of the regressions remain.

The regression summary below shows **inductor latency normalized to 3.7.2**, so higher is worse

## Regression summary (training)

23 cases more than 1.4x

| suite | dtype | name | inductor norm 3.7.2 | inductor norm 3.8.0+git9810c442 | inductor norm 3.8.0 (6c7855c1) |
|---|---|---|---|---|---|
| huggingface | amp_bf16 | BertForQuestionAnswering | 1.000 | 1.433 | 1.438 |
| huggingface | amp_bf16 | ElectraForQuestionAnswering | 1.000 | 1.593 | 1.591 |
| huggingface | amp_bf16 | LayoutLMForMaskedLM | 1.000 | 2.099 | 2.101 |
| huggingface | amp_bf16 | LayoutLMForSequenceClassification | 1.000 | 2.297 | 2.301 |
| huggingface | amp_bf16 | RobertaForQuestionAnswering | 1.000 | 1.430 | 1.434 |
| huggingface | amp_fp16 | AlbertForQuestionAnswering | 1.000 | 1.558 | 1.025 |
| huggingface | amp_fp16 | LayoutLMForMaskedLM | 1.000 | 2.004 | 2.004 |
| huggingface | amp_fp16 | LayoutLMForSequenceClassification | 1.000 | 2.194 | 2.195 |
| huggingface | bfloat16 | AlbertForMaskedLM | 1.000 | 1.521 | 1.007 |
| huggingface | bfloat16 | AlbertForQuestionAnswering | 1.000 | 1.854 | 1.007 |
| huggingface | bfloat16 | BertForQuestionAnswering | 1.000 | 1.444 | 1.443 |
| huggingface | bfloat16 | ElectraForQuestionAnswering | 1.000 | 1.676 | 1.675 |
| huggingface | bfloat16 | LayoutLMForMaskedLM | 1.000 | 2.153 | 2.152 |
| huggingface | bfloat16 | LayoutLMForSequenceClassification | 1.000 | 2.341 | 2.345 |
| huggingface | bfloat16 | MegatronBertForQuestionAnswering | 1.000 | 1.452 | 1.451 |
| huggingface | bfloat16 | RobertaForQuestionAnswering | 1.000 | 1.444 | 1.444 |
| huggingface | float16 | AlbertForQuestionAnswering | 1.000 | 1.575 | 0.999 |
| huggingface | float16 | LayoutLMForMaskedLM | 1.000 | 2.051 | 2.054 |
| huggingface | float16 | LayoutLMForSequenceClassification | 1.000 | 2.244 | 2.243 |
| timm_models | bfloat16 | mobilevit_s | 1.000 | 1.707 | 1.711 |
| timm_models | float16 | mobilevit_s | 1.000 | 1.891 | 1.885 |
| timm_models | float32 | mobilevit_s | 1.000 | 1.510 | 1.510 |
| torchbench | amp_fp16 | hf_T5_large | 1.000 | 1.452 | 1.454 |

Another ~ 100  cases (1.4x > latency > 1.10x)

All rows are `mode=training`, `scenario=performance`.

## Environment

- Triton good: `3.7.2`
- Triton bad: `3.8.0+git9810c442` and release `3.8.0 (6c7855c1)`
- Device: B60 Linux

## Reproducer
```bash
docker run -it -e TZ=Aisa/Shanghai --device=/dev/mem --device=/dev/dri --group-add video --privileged -v $(realpath ${HOME}):/home/jenkins --shm-size=8g intelgpu/ubuntu-26.04-rolling:26.18 bash

# setup python
curl -LsSf https://astral.sh/uv/install.sh | sh
source $HOME/.local/bin/env
uv venv myvenv --python 3.10 --clear
source myvenv/bin/activate
uv pip install pip wheel "setuptools<81"

uv pip install torch==2.13.0 torchaudio torchvision --index-url https://download.pytorch.org/whl/xpu

# prepare pytorch
pytorch_commit="$(python -c 'import torch; print(torch.version.git_version)')"
git clone https://github.com/pytorch/pytorch
cd pytorch
git checkout ${pytorch_commit}

# install deps
uv pip install pip pandas psutil scipy requests
uv pip install -r .ci/docker/ci_commit_pins/huggingface-requirements.txt
uv pip install timm

# sync model list
rm -rf torch torch-xpu-ops
git clone https://github.com/intel/torch-xpu-ops
rsync -avz ./torch-xpu-ops/.ci/benchmarks/ ./benchmarks/dynamo/

# install new triton
uv pip uninstall triton triton-xpu
rm -rf intel-xpu-backend-for-triton
git clone https://github.com/intel/intel-xpu-backend-for-triton
cd intel-xpu-backend-for-triton
git checkout 6c7855c1    # release/3.8.0
./scripts/compile-triton.sh

# run test
# example 1: BertForQuestionAnswering, amp_bf16, training
python benchmarks/dynamo/huggingface.py -d xpu --performance --amp --amp-dtype bfloat16 --training --backend=inductor --cold-start-latency -n10 --disable-cudagraphs --only BertForQuestionAnswering

# example 2: LayoutLMForSequenceClassification, bfloat16, training
python benchmarks/dynamo/huggingface.py -d xpu --performance --bfloat16 --training --backend=inductor --cold-start-latency -n10 --disable-cudagraphs --only LayoutLMForSequenceClassification

# example 3: mobilevit_s, float16, training
python benchmarks/dynamo/timm_models.py -d xpu --performance --float16 --training --backend=inductor --cold-start-latency -n10 --disable-cudagraphs --only mobilevit_s

```

**Comment by anmyachev** [comment]
Regression:
```bash
commit 24b2ed41f49a5406acd0697e167b17f11cf2e752
Author: Ettore Tiotto <ettore.tiotto@intel.com>
Date:   Wed Jun 24 16:59:53 2026 -0400

    [TritonIntelGPU] Add CanonicalizePointers pass (#7254)
    
    Decompose loop-carried `tensor<Nx!tt.ptr>` iter_args into scalar base +
    tensor offset to reduce GRF pressure. Enabled by default
    (`TRITON_INTEL_DISABLE_CANONICALIZE_POINTERS=1` to disable).
    
    The pass is ported from AMD's `CanonicalizePointers` and decomposes
    tensor pointer arithmetic into a `FatPtr{scalar_base, tensor_offset}`
    pair, hoisting the uniform (scalar-extractable) part of each offset onto
    the scalar base pointer. For pointer-heavy kernels this replaces
    loop-carried `tensor<Nx!tt.ptr>` iter_args with scalar `!tt.ptr`
    iter_args, cutting register pressure.
    
    Measured ~1.85x speedup on a fused multi-tensor sum microbenchmark on
    PVC (Data Center GPU Max 1100), with zero regressions in the Intel
    pytest suite.
    
    Closes #7252
    
    ---------
    
    Co-authored-by: Claude Opus 4.8 <noreply@anthropic.com>
```

@whitneywhtsang @chengjunlu @etiotto I don't think we have time to fix this performance regression so I prefer to disable this pass by default. Opinions?

**Comment by whitneywhtsang** [comment]
Let’s disable it in release branch.

**Comment by anmyachev** [comment]
The fix is in the `release/3.8.0` branch.

**Comment by anmyachev** [comment]
For `mobilevit_s` model, regression is still observed: https://github.com/intel/intel-xpu-backend-for-triton/issues/7520

**Comment by anmyachev** [comment]
> For `mobilevit_s` model, regression is still observed: [#7520](https://github.com/intel/intel-xpu-backend-for-triton/issues/7520)

Fixed.

**Comment by anmyachev** [comment]
@mengfei25 feel free to reopen if needed
